### PR TITLE
Added WebGL Earth API, geoip API, and gon gem

### PR DESCRIPTION
**Pull request by mailethiesen**
added WebGL Earth to shares#index and had javascript api link up to get user public ip, also added gon gem to get rails variables to pass into json format. Gon gem is hooked up in application.html.erb, the shares controller, and the shares index page.

WebGL Earth now appears on the shares index page with a marker at the users current location. The marker pop up tells the user what city they are in.

### CODE REVIEW

**Comment by mailethiesen** [comment]
The next step here would be to make a location for either each user or each share, then store the location with the share record. Then display each share on the map with it's own marker. Need to figure out how to either display multiple markers for one location or just display THE LAST SHARE from a given location. Not sure how to break that logic up.

File: app/views/shares/index.html.erb [hunk 1]
```diff
@@ -1,5 +1,50 @@
+    <style>
+      html, body{padding: 0; margin: 0;}
+      #earth_div{top: 0; right: 0; bottom: 0; left: 0; position: absolute !important;}
+    </style>
```
Comment (kenmazaika) [inline app/views/shares/index.html.erb]: We should move inline styles to a CSS file.

File: app/views/shares/index.html.erb [hunk 2]
```diff
@@ -1,5 +1,50 @@
...
+
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/2.1.4/jquery.min.js"></script>
```
Comment (kenmazaika) [inline app/views/shares/index.html.erb]: I believe we're already including jQuery.  I don't think this line is needed.

File: app/views/shares/index.html.erb [hunk 3]
```diff
@@ -1,5 +1,50 @@
...
+    <script src="http://www.webglearth.com/v2/api.js"></script>
+    <script>
+      function initialize() {
+          $.get( "http://geoip.nekudo.com/api/", function( data ) {
+            var latLon = [data.location.latitude, data.location.longitude];
+            var city = [data.city]
+            console.log( data );
+            console.log(latLon);
+            console.log(city);
+            console.log(gon.share_title);
```
Comment (kenmazaika) [inline app/views/shares/index.html.erb]: Might want to remove `console.log`'s

**Comment by kenmazaika** [comment]
Very cool.  Excited to see this.  Made a few comments and there look like there are merge conflicts that need tobe resolved.

**Comment by mailethiesen** [comment]
@kenmazaika where do I go to resolve the merge conflicts? I haven't done that before.

**Comment by kpat108** [comment]
You may need to merge the current master branch to your branch. You can pm me on slack if you need more help.

**Comment by kenmazaika** [comment]
Awesome to hear Kay can help you out.  In the mean time you may want to review this video:
http://www.thefirehoseproject.com/advanced-git-topics/5

**Pull request closed without merging**